numpy/ma/core.py

```diff
@@ -4286,22 +4297,43 @@ def count(self, axis=None):
         array([3, 0])
 
         """
+        kwargs = {} if keepdims is np._NoValue else {'keepdims': keepdims}
+
         m = self._mask
-        s = self.shape
+        # special case for matrices (we assume no other subclasses modify
+        # their dimensions)
```

Comment:
I honestly hope this is good, it would make life so much easier ;). But at assuming it at least within MA, seems like a reasonable choice to me and I am for giving it a shot.